Fix CVV Submission and Step Status Validation

bnahin opened the pull request:
This PR fixes the problems brought up in #2. It will accurately report the status of each checkout step as well as correctly send the CVV.

- Use the Result property to check for success rather than the status code.
- Fix the BillingProfile instantiation to correctly include the CVV
- Use the card provider from config rather than the hardcoded value
- Improve formatting
- Improve README
- Add GNU General Public license

MrTpat commented:
Mind getting rid of the formatting changes? It makes the diff much too large so I can't really focus on the code changes itself. For formatting, I'll put a linter in place later.

bnahin commented:
The actual changes are profiles.py:30 and communicator.py:133,119,99,94,87. And then, of course, the README and LICENSE.

The changes are rather simple; let me know if you still want me to remove the formatting and I'll make a new commit.

MrTpat commented on newegg/communicator.py:
```diff
@@ -79,38 +84,52 @@ def get_transaction_number(self, session_id: str, **kwargs) -> Optional[int]:
             DebugLogger.log(f'Submit order text: {req.text}')
             DebugLogger.log(f'Submit order status: {req.status_code}')
-            return req.status_code == 200
+            return req.json()["Result"] == "Success"
```
I think req.status_code == 200 and req.json()[["result"] check would be more comprehensive, could prevent parsing json() when we dont need to if its a 400 error.

MrTpat commented:
Fixed the cvv and card provider issue in my latest commits, Ill take your req["result"] == "success" PR once its properly rebased :)

MrTpat commented:
and the readme work too, that was huge

MrTpat commented:
Also we need to be careful doing the req["result"] == success check each time - did you verify whether or not those endpoints always return that message on success?

MrTpat commented:
And yeah if you could remove the formatting, I think that would be good, I want to avoid a monolithic commit with a non-defined-linter code style change

MrTpat commented:
Was able to confirm that the result being success only needs to be on the verify_address and the submit_order.

I fixed that just now as well, I believe that was all known issues there were, thanks for bringing them to my attention! when you have the time, can you update the PR so it only has the readme changes? I think that would be a valuable addition.

bnahin committed:
Improve README and add default configs

bnahin commented:
No problem! I have updated my branch. I also moved configs/ to configs-default/ so that the default configs have their own home for purposes of gitignoring.

bnahin commented:
I won't add it back in my fork but I would recommend adding the GNU General Public license.

MrTpat commented:
LGTM!
